scripts/microbench/microbench_parser.py

```diff
@@ -1,53 +1,150 @@
-import argparse
-
-parser = argparse.ArgumentParser(description='PyTorch models inference')
-parser.add_argument('-f', '--file', type=str, help='log file')
-parser.add_argument('-o', '--output', type=str, help='output file for parser results')
-args = parser.parse_args()
-success_ops = {}
-error_ops = {}
-skipped_ops = {}
-with open(args.file) as reader:
-    lines = reader.readlines()
-    op = ""
-    for line in lines:
-        if line.strip() == "":
-            continue
-        if line.startswith("Running"):
-            op = line.split(" ")[1].strip()
-        # Skipping aten.convolution_backward.default, no inductor impl
-        elif line.startswith("Skipping"):
-            op = line.split(",")[0].split(" ")[1].strip()
-            reason = line.split(",")[1].strip()
-            if op not in skipped_ops.keys():
-                skipped_ops[op] = reason
-                # print(op+", " + reason)
-        # error aten.stack.default
-        elif line.startswith("error"):
-            op = line.split(" ")[1].strip()
-            if op not in error_ops.keys():
-                error_ops[op] = "error"
-                # print(op + ", error")
-        if line.startswith("Inductor"):
-            speedups = line.split("[")[-1].split("]")[0]
-            if op not in success_ops.keys():
-                success_ops[op] = speedups
-                # print(op+", "+speedups)
-
-if (args.output):
-    with open(args.output, 'w') as writer:
-        writer.write("op_name, speedup_0.2, speedup_0.5, speedup_0.8\n")
-        for op in sorted(success_ops):
-            writer.write(op + ", " + success_ops[op] + "\n")
-        for op in sorted(skipped_ops):
-            writer.write(op + ", " + skipped_ops[op] + "\n")
-        for op in sorted(error_ops):
-            writer.write(op + ", " + error_ops[op] + "\n")
-else:
-    print("op_name, speedup_0.2, speedup_0.5, speedup_0.8")
-    for op in sorted(success_ops):
-        print(op + ", " + success_ops[op])
-    for op in sorted(skipped_ops):
-        print(op + ", " + skipped_ops[op])
-    for op in sorted(error_ops):
-        print(op + ", " + error_ops[op])
+"""microbench_parser.py
+Generate html report.
+Usage:
+  python microbench_parser.py -w 20221228 -l https://inteltf-jenk.sh.intel.com/job/inductor_dashboard/100/ -n mlp-validate-icx24-ubuntu
+"""
+
+import argparse
+import os
+import subprocess
+import pandas as pd
+from pandas import ExcelWriter
+
+parser = argparse.ArgumentParser(description='PyTorch models inference')
+parser.add_argument('-w', '--workday', type=str, help='workday of refresh')
+parser.add_argument('-l', '--url', type=str, help='jenkins build url')
+parser.add_argument('-n', '--node', type=str, help='jenkin node lable')
+args = parser.parse_args()
+
+path=os.getcwd()

```

Comment:
Does it means this scripts just can execute in log dir? Can we change the hardcode log path as a parameter? Instead of `-w, --workday`, we can use `-o, --output` to include the workday info if we need. We'd better to make our scripts (except groovy scripts) can be used for others, not for automation test only. Those changes can help to make the scripts pure enough

Reply:
Thanks. I add `--log_path` for specify log path, use `os.getcwd()` when this parameter not provided. Replace `-w` also.